src/main/java/community/solace/spring/integration/leader/leader/LeaderEventDebouncer.java

```diff
@@ -0,0 +1,73 @@
+package community.solace.spring.integration.leader.leader;
+
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.integration.leader.Context;
+import org.springframework.integration.leader.event.DefaultLeaderEventPublisher;
+import org.springframework.integration.leader.event.LeaderEventPublisher;
+
+import java.time.Duration;
+import java.util.concurrent.*;
+
+public class LeaderEventDebouncer implements LeaderEventPublisher, AutoCloseable {
+    private final LeaderEventPublisher leaderEventPublisher;
+    private final ScheduledExecutorService scheduler;
+    private final Duration delay;
+    private final ConcurrentHashMap<String, ScheduledFuture<?>> pendingTasks = new ConcurrentHashMap<>();
+
+    LeaderEventDebouncer(ApplicationEventPublisher applicationEventPublisher, Duration delay) {
+        this.leaderEventPublisher = new DefaultLeaderEventPublisher(applicationEventPublisher);
+        this.delay = delay;
+        this.scheduler = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    @Override
+    public void publishOnGranted(Object source, Context context, String role) {
+        scheduleEvent(role, () ->
+                leaderEventPublisher.publishOnGranted(source, context, role));
+    }
+
+    @Override
+    public void publishOnRevoked(Object source, Context context, String role) {
+        scheduleEvent(role, () ->
+                leaderEventPublisher.publishOnRevoked(source, context, role));
+    }
+
+    @Override
+    public void publishOnFailedToAcquire(Object source, Context context, String role) {
+        scheduleEvent(role, () ->
+                leaderEventPublisher.publishOnFailedToAcquire(source, context, role));
+    }
+
+    private void scheduleEvent(String role, Runnable task) {
+        ScheduledFuture<?> existing = pendingTasks.get(role);
```

Comment:
since we realy care about consistency I strongly suggest to do actual locking with syncronized instead of using concurrent objects